### ISSUE TITLE
Integrate Mediawiki web-fonts

### ISSUE THREAD

**Issue by kelson42** [kiwix/mwoffliner#19]
Should work out-of-the-box if resourceLoader works offline... but to be checked.
First reported at https://sourceforge.net/p/kiwix/feature-requests/333/

**Comment by kelson42** [comment]
See this thread which bring a few information https://lists.wikimedia.org/pipermail/mediawiki-l/2017-April/046504.html

**Comment by kelson42** [comment]
Maybe being connected using a user were the fonts would be activated would work!?

**Comment by ninjastrikers** [comment]
Any update?

**Comment by kelson42** [comment]
@ninjastrikers Not really, so far it is not clear how to solve the full problem. Mediawiki/Wikipedia seems to have abandonned the web font solution...

**Comment by amire80** [comment]
Hi,

I was asked to comment on this.

I am happy to see that there is interest in this, because it makes a lot of sense.

@santhoshtr may have more details, but I'll write what I know.

https://github.com/wikimedia/jquery.webfonts/ isn't abandoned. It's not actively developed, but it does receive maintenance updates. The corresponding MediaWiki extension is UniversalLanguageSelector, and it is regularly maintained, too.

Webfonts are available on all Wikimedia projects, but on most of them they are disabled by default for performance and bandwidth reasons. Every user can enable them, and on some projects they are enabled by default if the editors community requested it. These performance issues are not supposed to be a problem for you, because if I understand correctly, your users download everything once and then work offline, and the fonts files are tiny when compared to the huge Wikipedia dumps that they download.

I don't know how exactly does your project work. You may want to use the standalone jquery library, or the whole UniversalLanguageSelector extension, or maybe just the fonts library that we compiled. Quite a lot of attention was dedicated to making it work well in different browsers—see the documentation files in jquery.webfonts for that.

If there's any more information I can provide, please let me know.

**Comment by kelson42** [comment]
@amire80 Thank you for your feedback. Do you know what would be the way through the API to know if there is such a webfont available for a Mediawiki instance? And if "yes" how to get the Javascript machinery to get it loaded properly?

**Comment by amire80** [comment]
I'm not sure it's available through the HTTP MediaWiki action API. If the user enabled the option, then the definedness of the JavaScript `$.webfonts` variables is a sign that the feature is enabled. In any case, if the UniversalLanguageSelector extension is installed, the JavaScript variable `$.uls` will be defined. As I mentioned earlier, the extension is installed in all Wikimedia projects, even if the webfonts feature is disabled by default. Perhaps @santhoshtr has a better answer.

**Comment by santhoshtr** [comment]
The webfonts we use in Wikipedia sites is based on this configuration https://github.com/wikimedia/mediawiki-extensions-UniversalLanguageSelector/blob/master/resources/js/ext.uls.webfonts.repository.js You can see per language fonts. The font files are present in the same repo.

Depending on the architecture, you may either use jquery.webfonts with this repository(It takes this repository as a configuration and applies fonts based on lang attribtues) or use a simple css @font-face rules. They are very powerful and flexible now a days compared to the days when we wrote jquery.webfonts.
